libs/cases/frontend/view/src/lib/components/create-case/create-case.component.ts

```diff
@@ -1,70 +1,129 @@
-
-	wasValidated = false;
-	value = '1';
-
-	createCaseForm = {
-		selectFacility: '',
-		phone: '',
-		email: '',
-		text: '',
-	};
-
-	onSubmit(form: NgForm): void {
-		this.wasValidated = true;
-		if (form.form.valid) {
-			//Form is valid!!
-			//get all values by form.form.value
-		}
-	}
-
-	public setFacilityValue(value: string) {
-		this.createCaseForm.selectFacility = value;
-	}
-
-	public setPhoneValue(value: string) {
-		this.createCaseForm.phone = value;
-	}
-
-	public setEmailValue(value: string) {
-		this.createCaseForm.email = value;
-	}
-
-	public getFacilityValue(): IFacilityMock | undefined {
-		return this.facilities.find(
-			(facility) => facility.id === this.createCaseForm.selectFacility,
-		);
-	}
-
-	public getPhoneValue() {
-		return this.createCaseForm.phone;
-	}
-
-	public getEmailValue() {
-		return this.createCaseForm.email;
-	}
-
-	public getTextValue() {
-		return this.createCaseForm.text;
-	}
+    private readonly _browseFacade = inject(XdBrowseFacade);
+    protected readonly _browseFacade2 = inject(XdBrowseFacadesService);
+    protected readonly facilities = toSignal(this._browseFacade.getAllTimeseries());
+
+    constructor(private readonly toastService: ToastService) {
+    }
+
+    casePriority = CasePriority;
+    caseType = CaseType;
+    wasValidated = false;
+    value = '1';
+
+    createCaseForm = {
+        selectFacility: '',
+        title: '',
+        dueDate: '',
+        selectPriority: '',
+        selectType: '',
+        phone: '',
+        email: '',
+        text: '',
+    };
+
+    /**
+     * called when the user presses the Create Case Button
+     */
+    onSubmit(form: NgForm): void {
+        this.wasValidated = true;
+
+        if (form.valid) {
+            const caseData = this.mapFormData(form.form.value);
+
+            this._browseFacade2.createCase(caseData).subscribe({
+                // eslint-disable-next-line @typescript-eslint/no-unused-vars
+                next: (_) => {
+                    this.showSuccessToast();
+                },
+            });
+        }
+    }
+
+    async showSuccessToast() {
+        await this.toastService.show({
+            type: "success",
+            message: 'Successfully created Case'
+        });
+    }
+
+    public setFacilityValue(value: string) {
+        this.createCaseForm.selectFacility = value;
+    }
+
+    public setPhoneValue(value: string) {
+        this.createCaseForm.phone = value;
+    }
+
+    public setEmailValue(value: string) {
+        this.createCaseForm.email = value;
+    }
+
+    public getFacilityValue() {
+        return this.facilities()?.find(
+            (facility) => facility.id === this.createCaseForm.selectFacility,
+        );
+    }
+
+    public getPhoneValue() {
+        return this.createCaseForm.phone;
+    }
+
+    public getEmailValue() {
+        return this.createCaseForm.email;
+    }
+
+    public getTextValue() {
+        return this.createCaseForm.text;
+    }
```

Comment:
1. If you want to use encapsulation to set the values via template driven approach and *not* directly use *dynamicForms* and use encapsulation. You need to use `get `and `set `here as keyword: https://www.typescripttutorial.net/typescript-tutorial/typescript-getters-setters/

How do use dynamic forms (in my perception better with so many validators)
https://angular.dev/guide/forms/dynamic-forms

2. Some quotes are in `"` Reformat.

Reply:
At the beginning I wanted to use the dynamic Forms, but Ix uses template driven Forms: https://ix.siemens.io/docs/controls/validation